NextcloudTalk/LoginViewController.swift

```diff
@@ -0,0 +1,334 @@
+        qrCodeButton.setTitle(NSLocalizedString("Scan QR code", comment: ""), for: .normal)
+        qrCodeButton.isHidden = !QRScannerViewController.isDataScannerSupported()
+
+        // Import account button
+        importAccountButton.setTitle(NSLocalizedString("Import account", comment: ""), for: .normal)
+
+        // Buttons style
+        [loginButton, qrCodeButton, importAccountButton].forEach { button in
+            button.backgroundColor = NCAppBranding.brandColor()
+            button.layer.borderColor = NCAppBranding.brandTextColor().cgColor
+            button.layer.cornerRadius = 12
+            button.layer.borderWidth = 1
+            button.clipsToBounds = true
+
+            // Highlighted style
+            button.setTitleColor(NCAppBranding.brandTextColor().withAlphaComponent(0.5), for: .highlighted)
+            if let buttonImage = button.image(for: .normal) {
+                button.setImage(buttonImage.withTintColor(NCAppBranding.brandTextColor().withAlphaComponent(0.5), renderingMode: .alwaysOriginal), for: .highlighted)
+            }
+        }
+
+        // Activity indicator
+        activityIndicatorView.color = NCAppBranding.brandTextColor()
+        activityIndicatorView.isHidden = true
+
+        // Cancel button
+        cancelButton.setTitle(NSLocalizedString("Cancel", comment: ""), for: .normal)
+        cancelButton.setTitleColor(NCAppBranding.brandTextColor(), for: .normal)
+        cancelButton.isHidden = !(multiAccountEnabled.boolValue && NCDatabaseManager.sharedInstance().numberOfAccounts() > 0)
+
+        // Check for Nextcloud Files app accounts
+        checkFilesAppAccounts()
+
+        // Add tap gesture recognizer to dismiss keyboard
+        view.addGestureRecognizer(
+            UITapGestureRecognizer(target: self, action: #selector(dismissKeyboard))
+        )
+    }
+
+    override var preferredStatusBarStyle: UIStatusBarStyle {
+        return NCAppBranding.statusBarStyleForBrandColor()
+    }
+
+    override var supportedInterfaceOrientations: UIInterfaceOrientationMask {
+        return .portrait
+    }
+
+    // MARK: - Certificate delegate
+
+    func trustedCerticateAccepted() {
+        loginButtonPressed(self)
+    }
+
+    // MARK: - UITextField delegate
+
+    func textFieldShouldReturn(_ textField: UITextField) -> Bool {
+        textField.resignFirstResponder()
+        startLoginProcess()
+        return true
+    }
+
+    func dismissKeyboard() {
+        view.endEditing(true)
+    }
+
+    // MARK: - Login
+
+    func startLoginProcess() {
+        guard let serverTextFieldText = serverTextField.text?.trimmingCharacters(in: .whitespaces),
+              !serverTextFieldText.isEmpty else {
+            serverTextField.becomeFirstResponder()
+            return
+        }
+
+        startLoginProcess(serverURL: serverTextFieldText, user: nil)
+    }
+
+    func startLoginProcess(serverURL: String, user: String?) {
+        var normalizedServerURL = serverURL
```

Comment:
```suggestion
        var normalizedServerURL = serverURL.lowercased()
```

to make sure the next check with `hasPrefix` works in all cases?